src/transaction/log_recovery.c

```diff
@@ -809,7 +807,14 @@ log_recovery (THREAD_ENTRY * thread_p, int ismedia_crash, time_t * stopat)
   if (!context.is_page_server ())
     {
       log_recovery_undo (thread_p);
-      boot_reset_db_parm (thread_p);
+
+      /* Flush all dirty pages */
+      logpb_flush_pages_direct (thread_p);
+      logpb_flush_header (thread_p);
+      (void) pgbuf_flush_all (thread_p, NULL_VOLID);
+
+      // Reset boot_Db_parm in case a data volume creation was undone.
+      (void) boot_reset_db_parm (thread_p);
```

Comment:
Although this was the legacy behavior, I think we can change and remove the flush after undo completely. Checkpoint is a bit later log/data pages are flushed again. This is overkill.

I would even argue that checkpoint/flush during recovery is unnecessary, but that's a different matter, outside the scope of this patch.

```suggestion
      // Reset boot_Db_parm in case a data volume creation was undone.
      (void) boot_reset_db_parm (thread_p);
```

Reply:
What about `log_flush_pages_direct` after `log_recovery_redo` in `log_recovery_finalize_transactions`?